Utility for filtering and parsing log messages

krzysztofx opened the pull request:
This utility simplifies filtering and parsing specific messages
from given log file. It tracks log file using utils_tail_match.h API.
Main feature is automatic messages assembly based on set of user provided
regular expressions containing distinguished expressions for recognizing
start and the end of the message. That makes assembling single message
from multiple log lines quite easy. Utility supports multiple parsing jobs
and also user defined message part validation flags (mandatory or not).

Added fixes for memory leak in utils_match.c and for debug mode
segfault in src/utils_tail_match.c

Modification of utils_tail and utils_tail_match API with adding option
for reading file from the beginning. Minor modifications in this API
calls in tail and tail_csv plugins.

pszczerx commented:
@rubenk, @octo any comments? We have more patches which depend on this utility. Could I ask for review?

rubenk commented:
@pszczerx you do know we both have a day job right? ;) We will get to your PR eventually.

rubenk commented:
bbb605b looks like a legit bug fix, any change you could open a separate PR for that, against the oldest branch it applies to?

krzysztofx commented:
@rubenk 

Regex related memory leak is present even in very first version of utils_match.c , but  the oldest existing branch containing it  is collectd-4.10, so you want me to open PR for it ?

krzysztofx commented:
created PR https://github.com/collectd/collectd/pull/2182

krzysztofx commented:
update, PR https://github.com/collectd/collectd/pull/2182 for collectd 4.10 removed. Memory leak fix backported for collectd 5.6 with PR https://github.com/collectd/collectd/pull/2192

rubenk commented:
Please rebase and drop bbb605b

krzysztofx commented:
Rebased , requested commit [bbb605b](https://github.com/collectd/collectd/commit/bbb605b2a4d637ff66d433ffb0fd643ee61deb34l) removed  and added some bugfix in [db6aef7
](https://github.com/collectd/collectd/pull/2154/commits/db6aef7dc087f6927f2c8090f372cfb6d8768ee5)  I see following errors on epel6 platforms:

    src/daemon/common.c: In function 'check_capability':
    src/daemon/common.c:1571: error: implicit declaration of function 'CAP_IS_SUPPORTED'
    make[1]: *** [src/daemon/common.lo] Error 1

Those started occurring  even earlier, see : [build 1403 ](https://ci.collectd.org/job/pull-requests-github_trigger-aggregation/1403/) . 
Is that known issue ? Unfortunatelly I don't have access to any epel6 to investigate it.

Thanks
Krzysztof

rubenk commented:
Thanks, the build breakage seems unrelated so you can ignore this. @mfournier would you mind taking a look at build 1403? seems related to the capability change.

krzysztofx commented:
@rubenk @octo 
Guys, forgive me, but this is another  annoying request. Is there anything we could do to get this utility merged in near future ?  Can we somehow contribute  to this review? I wouldn't push , but we have queued  at least two plugins/extensions that rely on this parser. It's mcelog plugin extension  for all MCE's , not only memory related  and another one is plugin for collecting PCIe errors.  Since this parser is still pending in review with no feedback,  we're a bit uncertain if  this work  will be accepted by community and if we should invest more effort on developing other plugins that utilize this parser.  

thanks
Krzysztof

krzysztofx commented:
Rebased to latest master branch

mrunge commented:
Any update here?
